include/oneapi/dpl/pstl/hetero/dpcpp/utils_ranges_sycl.h

```diff
@@ -231,8 +231,19 @@ struct is_passed_directly<zip_iterator<Iters...>> : ::std::conjunction<is_passed
 {
 };
 
+template <typename Iter, typename Void = void>
+struct is_passed_directly_legacy_support : ::std::false_type
+{
+};
+
+template <typename Iter> // for permutation_iterators
+struct is_passed_directly_legacy_support<Iter, ::std::enable_if_t<Iter::is_passed_directly::value>> : ::std::true_type
+{
+};
```

Comment:
I would call it ``is_passed_directly_legacy_trait``.

Why does the comment mention permutation iterators?

Reply:
Name changed, thanks.

I don't think `std::conditional` simplifies anything, because we need to handle 3 cases for `is_passed_directly`, `true_type`, `false_type` and missing.  

The comment was made in error, good catch.  A permutation iterator was triggering the error that hit this bug, but its used for more than just that.  I removed it.